### PR TITLE
Add lilac docker image and use a lighter 5x image for HF

### PR DESCRIPTION
**Pull request by dsmilkov**
CleanShot 2023-10-09 at 13 54 23@2x

Fixes https://github.com/lilacai/lilac/issues/737

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ENV PYTHONUNBUFFERED True
+
+# Adds GCC and other build tools so we can compile hnswlib and other native/C++ deps.
+RUN apt-get update --fix-missing && apt-get install -y --fix-missing build-essential && \
+  rm -rf /var/lib/apt/lists/*
+
+# Set the working directory in the container.
+WORKDIR /app
+
+RUN python -m pip install lilac[all]
```
Comment (nsthorat) [inline Dockerfile:13]: maybe comment here about optional deps and reducing size

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# Install from the local wheel inside ./dist. This will be a no-op if the wheel is not found.
+COPY --chown=user /dis[t] ./dist/
+RUN python -m pip install --find-links=dist --upgrade lilac[all]
+
+COPY LICENSE .
+
+EXPOSE 8000
```
Comment (nsthorat) [inline Dockerfile:21]: arg here? https://docs.docker.com/engine/reference/builder/#arg

**Review by nsthorat — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (nsthorat) [reply Dockerfile:21]: nevermind, you clarified that this is just for exposing & documentation

File: Dockerfile (hunk 1 above)
Comment (dsmilkov) [reply Dockerfile:14]: done

**Pull request merged**